src/features/layers_in_area/atoms/areaLayersDetailsResource/areaLayersDetailsParamsAtom.ts

```diff
@@ -30,13 +30,27 @@ export const areaLayersDetailsParamsAtom = createAtom(
 
     const mustBeRequested = allLayers.filter((layer) => {
       const isEnabled = enabledLayers.has(layer.id);
-      if (isEnabled) {
-        const cacheKey = layer.eventIdRequiredForRetrieval ? eventId : null;
-        const cached = cache.get(cacheKey)?.get(layer.id) ?? null;
+      if (!isEnabled) return false;
+
+      if (layer.boundaryRequiredForRetrieval) {
+        const cacheKey: string | undefined = focusedGeometry?.geometry['hash'];
```

Comment:
instead of `geometry['hash']` better to align type in
`src/core/shared_state/focusedGeometry.ts`
```ts
export interface FocusedGeometry {
  source: GeometrySource;
  geometry: GeoJSON.GeoJSON & { hash: string };
}
```

Reply:
I worried it will lead to big refactoring, but it looks much more clearer in the end 👍